src/github/types/config.ts

```diff
@@ -2,32 +2,58 @@ import { Type as T } from "@sinclair/typebox";
 import { StaticDecode } from "@sinclair/typebox";
 import { githubWebhookEvents } from "./webhook-events";
 
-enum Commands {
-  Start = "start",
-  Stop = "stop",
+const pluginNameRegex = new RegExp("^([0-9a-zA-Z-._]+)/([0-9a-zA-Z-._]+)(?::([0-9a-zA-Z-._]+))?(?:@([0-9a-zA-Z]+))?$");
+
+type GithubPlugin = {
+  owner: string;
+  repo: string;
+  workflowId: string;
+  ref?: string;
+};
+
+function githubPluginType() {
+  return T.Transform(T.String())
+    .Decode((value) => {
+      const matches = value.match(pluginNameRegex);
+      if (!matches) {
+        throw new Error(`Invalid plugin name: ${value}`);
+      }
+      return {
+        owner: matches[1],
+        repo: matches[2],
+        workflowId: matches[3] || "compute.yml",
```

Comment:
We will default to `action.yml` in the project root of the plugin. See my conversation rewards(work in progress) as an example: https://github.com/ubiquibot/conversation-rewards/blob/development/action.yml

This is to conform to the existing approach that GitHub Actions takes, and to enable compatibility with all existing GitHub Actions.

Reply:
I don't think you can invoke workflows in the root directory because those are meant to be published to the marketplace and used in a workflow.

Reply:
We can just make it a pass through action. Basically the only logic will be to invoke a workflow in the workflow folder.

Reply:
I still think we should default to `action.yml` in the root. 

@whilefoo @gentlementlegen rfc